controllers/controller.go

```diff
@@ -79,6 +92,8 @@ func (r *EtcdadmClusterReconciler) SetupWithManager(ctx context.Context, mgr ctr
 	r.controller = c
 	r.recorder = mgr.GetEventRecorderFor("etcdadm-cluster-controller")
 	r.uncachedClient = mgr.GetAPIReader()
+	r.SetIsPortOpen(isPortOpen)
+	r.GetEtcdClient = r.generateEtcdClient
```

Comment:
Should this be protected by `if == nil`?

Reply:
This func is only called in main right now and is not set there. But yeah I can add it as a failsafe